watertap/property_models/ion_DSPMDE_prop_pack.py

```diff
@@ -0,0 +1,838 @@
+                    if value(var[ind]) != val:
+                        raise ConfigurationError(
+                            "While using the calculate_state method on {sb_name}, {v_name} was "
+                            "fixed to a value {val}, but it was already fixed to value {val_2}. "
+                            "Unfix the variable before calling the calculate_state "
+                            "method or update var_args."
+                            "".format(sb_name=sb.name, v_name=var.name, val=val, val_2=value(var[ind])))
+                else:
+                    flags[(k, v_name, ind)] = False
+                    var[ind].fix(val)
+
+            if degrees_of_freedom(sb) != 0:
+                raise RuntimeError("While using the calculate_state method on {sb_name}, the degrees "
+                                   "of freedom were {dof}, but 0 is required. Check var_args and ensure "
+                                   "the correct fixed variables are provided."
+                                   "".format(sb_name=sb.name, dof=degrees_of_freedom(sb)))
+
+        # Solve
+        with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+            results = solve_indexed_blocks(opt, [self], tee=slc.tee)
+            solve_log.info_high("Calculate state: {}.".format(idaeslog.condition(results)))
+
+        if not check_optimal_termination(results):
+            _log.warning("While using the calculate_state method on {sb_name}, the solver failed "
+                         "to converge to an optimal solution. This suggests that the user provided "
+                         "infeasible inputs, or that the model is poorly scaled, poorly initialized, "
+                         "or degenerate.")
+
+        # unfix all variables fixed with var_args
+        for (k, v_name, ind), previously_fixed in flags.items():
+            if not previously_fixed:
+                var = getattr(self[k], v_name)
+                var[ind].unfix()
+
+        # fix state variables if hold_state
+        if hold_state:
+            fix_state_vars(self)
+
+        return results
+
+@declare_process_block_class("DSPMDEStateBlock",
+                             block_class=_DSPMDEStateBlock)
+class DSPMDEStateBlockData(StateBlockData):
+    def build(self):
+        """Callable method for Block construction."""
+        super(DSPMDEStateBlockData, self).build()
+
+        self.scaling_factor = Suffix(direction=Suffix.EXPORT)
+
+        # Add state variables
+        self.flow_mol_phase_comp = Var(
+            self.params.phase_list,
+            self.params.component_list,
+            initialize=100, #todo: revisit
+            bounds=(1e-8, None),
+            domain=NonNegativeReals,
+            units=pyunits.mol/pyunits.s,
+            doc='Mole flow rate')
+
+        self.temperature = Var(
+            initialize=298.15,
+            bounds=(273.15, 373.15),
+            domain=NonNegativeReals,
+            units=pyunits.degK,
```

Comment:
Minor point, but you can use just `K` here (and it is technically more correct - for whatever reason, Kelvins are not meant to be used with "degrees").

Reply:
Will do.